substrate/utils/frame/benchmarking-cli/src/pallet/logging.rs

```diff
@@ -0,0 +1,75 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::LOG_TARGET;
+use log::LevelFilter;
+use sp_core::{LogLevelFilter, RuntimeInterfaceLogLevel};
+use sp_runtime_interface::{
+	pass_by::{PassAs, PassFatPointerAndRead, ReturnAs},
+	runtime_interface,
+};
+use std::cell::OnceCell;
+
+thread_local! {
+	/// Log level that the runtime will use.
+	///
+	/// Must be initialized by the host before invoking the runtime executor.
+	static RUNTIME_LOG_LEVEL: OnceCell<LevelFilter> = OnceCell::new();
+}
+
+/// Init runtime logger with the following priority (high to low):
+/// - CLI argument
+/// - Environment variable
+/// - Default logger settings
+pub fn init(arg: Option<LevelFilter>) {
+	let level = arg.unwrap_or_else(|| {
+		if let Ok(env) = std::env::var("RUNTIME_LOG") {
```

Comment:
Did you tried this? From the code, it seems that `FromStr` for `LevelFilter` is only for parsing `Debug` etc and not `something=debug`?